lib/std/fmt/parse_hex_float.zig

```diff
@@ -202,18 +202,21 @@ pub fn parseHexFloat(comptime T: type, s: []const u8) !T {
         exponent += 1;
     }
 
-    // There are two cases to handle:
-    // - We've truncated more than 0.5ULP (R=S=1), increase the mantissa.
-    // - We've truncated exactly 0.5ULP (R=1 S=0), increase the mantissa if the
-    //   result is odd (G=1).
-    // The two checks can be neatly folded as follows.
-    mantissa |= @boolToInt(mantissa & 0b100 != 0);
```

Comment:
Changing this line to:
mantissa += @boolToInt(mantissa & 0b100 != 0);

seems to fix the issue without needing to check both sides of the truncation.

Reply:
and then removing the next line: `mantissa += 1`

Reply:
I did a cleanup commit + merge. If you have an additional improvement to make, would you mind submitting it as a follow-up PR?